src/lamp_py/bus_performance_manager/vehicle_events.py

```diff
@@ -0,0 +1,133 @@
+import re
+from datetime import timedelta, date
+from typing import List, Tuple, Dict
+
+from lamp_py.aws.s3 import (
+    file_list_after,
+    file_list_from_s3,
+    get_datetime_from_partition_path,
+    get_last_modified_object,
+)
+
+from lamp_py.runtime_utils.remote_files import RemoteFileLocations
+
+
+class BusInputFiles:
+    """
+    small dataclass to keep track of vp files and tm files for a given service
+    date
+    """
+
+    def __init__(self, service_date: date):
+        self.service_date: date = service_date
+        self.vp_files: List[str] = []
+        self.tm_files: List[str] = []
+
+
+def get_new_event_files() -> Tuple[List[str], List[str]]:
+    """
+    get new realtime files from the vehicle position and transit master s3
+    locations that have been populated since the last time vehicle events were
+    processed.
+    """
+    latest_event_file = get_last_modified_object(
+        bucket_name=RemoteFileLocations.bus_events.bucket_name,
+        file_prefix=RemoteFileLocations.bus_events.file_prefix,
+        version="1.0",
+    )
+
+    if latest_event_file:
+        cutoff = latest_event_file["last_modified"] - timedelta(hours=1)
+        vp_files = file_list_after(
+            bucket_name=RemoteFileLocations.vehicle_positions.bucket_name,
+            file_prefix=RemoteFileLocations.vehicle_positions.file_prefix,
+            cutoff=cutoff,
+        )
+
+        tm_files = file_list_after(
+            bucket_name=RemoteFileLocations.tm_stop_crossing.bucket_name,
+            file_prefix=RemoteFileLocations.tm_stop_crossing.file_prefix,
+            cutoff=cutoff,
+        )
+
+    else:
+        vp_files = file_list_from_s3(
+            bucket_name=RemoteFileLocations.vehicle_positions.bucket_name,
+            file_prefix=RemoteFileLocations.vehicle_positions.file_prefix,
+        )
+
+        tm_files = file_list_from_s3(
+            bucket_name=RemoteFileLocations.tm_stop_crossing.bucket_name,
+            file_prefix=RemoteFileLocations.tm_stop_crossing.file_prefix,
+        )
+
+    return vp_files, tm_files
+
+
+def files_per_service_date(
+    vp_files: List[str], tm_files: List[str]
+) -> Dict[date, BusInputFiles]:
+    """
+    take a list of input files and bucket them based on the service date they
+    cover.
+    """
```

Comment:
I believe there could be a flaw with the methodology of this function where the `get_new_event_files` function picks up `vp_files` for a certain service date, but no `tm_files` are returned for that service date (because the files aren't available based on the `cutoff` parameter. 

For the next step of this process to work correctly I believe we will need all `vp_files` and `tm_files` for a specific `service_date` whether the input files were modified or not. 

An alternate approach could be determining the list of `service_date`s that need to be processed based on `get_last_modified_object` and then pulling all available `vp_files` and `tm_files` for each of those service dates.

Reply:
i think that makes sense.

* pull service dates from file names
* another round of s3 calls to get all files for those service dates
* return the maps of service dates to files.